src/io/utilities/readSBML.m

```diff
@@ -636,10 +640,10 @@
     model.lb = fbc_lb;
     model.ub = fbc_ub;
     if noObjective==0; % when there is an objective function
-        indexObj=findRxnIDs(model,fbc_obj);
+        indexObj=findRxnIDs(model,cleanUpFormatting(fbc_obj));
         % indexObj=find(strcmp(fbc_obj,model.rxns))
-        model.c(indexObj)=1;
-        model.osense = - sign(fbc_obj_value);
+        model.c(indexObj) = fbc_obj_value(:);
+        model.osense = fbc_obj_sense;
```

Comment:
@shjchan, thanks for willing to help! :tada:

You can create a new PR, or you can push to the branch `pr-375` on my fork (I invited temporarily to collaborate). You simply have to do:
````sh
git clone https://github.com/laurentheirendt/cobratoolbox.git
git checkout pr-375
# do your changes
git add [yourFiles]
git commit -m "[yourMessage]"
git push origin pr-375
````